src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs

```diff
@@ -1,19 +1,14 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
@iSazonov, your last commit had 4 failures in `PowerShell-CI-macos`
Can load a native assembly.Can load native dll
```powershell
Expected $null or empty, but got /Users/runner/runners/2.160.1/work/1/a/bins/publish\libmi.dylib.
at <ScriptBlock>, /Users/runner/runners/2.160.1/work/1/s/test/powershell/engine/Basic/Assembly.LoadNative.Tests.ps1: line 68
68:         "${PSHOME}\${sourceDllName}" | Should -BeNullOrEmpty
```
Can load a native assembly.Can load native dll
```powershell
Expected $null or empty, but got /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp1QrqVn.tmp/testDllNativeFolder/osx-x64;/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp1QrqVn.tmp/testDllNativeFolder/managed.dll.
at <ScriptBlock>, /Users/runner/runners/2.160.1/work/1/s/test/powershell/engine/Basic/Assembly.LoadNative.Tests.ps1: line 72
72:         (dir $root) -join ";" | Should -BeNullOrEmpty
```
Can load a native assembly.Can load native dll
```powershell
Expected $null or empty, but got /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp1QrqVn.tmp/testDllNativeFolder/osx-x64/nativedll.dylib.
at <ScriptBlock>, /Users/runner/runners/2.160.1/work/1/s/test/powershell/engine/Basic/Assembly.LoadNative.Tests.ps1: line 76
76:         (dir $archFolder) -join ";" | Should -BeNullOrEmpty
```
Can load a native assembly.Can load native dll
```powershell
Exception calling "LoadNative" with "0" argument(s): "Unable to load shared library '/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp1QrqVn.tmp/osx-x64/nativedll.dylib' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable: dlopen(/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp1QrqVn.tmp/osx-x64/nativedll.dylib, 1): image not found"
at <ScriptBlock>, /Users/runner/runners/2.160.1/work/1/s/test/powershell/engine/Basic/Assembly.LoadNative.Tests.ps1: line 84
84:         [TestNativeClass2]::LoadNative() | Should -Not -BeNullOrEmpty

```